sealedenumprocessor/src/test/kotlin/com/livefront/sealedenum/EnvironmentsSealedEnum.kt

```diff
@@ -0,0 +1,159 @@
+        Environments.Https.Google
+    )
+
+
+    override val enumClass: Class<EnvironmentsEnum>
+        get() = EnvironmentsEnum::class.java
+
+    override fun ordinalOf(obj: Environments): Int = when (obj) {
+        Environments.Http.Livefront -> 0
+        Environments.Http.Google -> 1
+        Environments.Https.Livefront -> 2
+        Environments.Https.Google -> 3
+    }
+
+    override fun nameOf(obj: Environments): String = when (obj) {
+        Environments.Http.Livefront -> "Environments_Http_Livefront"
+        Environments.Http.Google -> "Environments_Http_Google"
+        Environments.Https.Livefront -> "Environments_Https_Livefront"
+        Environments.Https.Google -> "Environments_Https_Google"
+    }
+
+    override fun valueOf(name: String): Environments = when (name) {
+        "Environments_Http_Livefront" -> Environments.Http.Livefront
+        "Environments_Http_Google" -> Environments.Http.Google
+        "Environments_Https_Livefront" -> Environments.Https.Livefront
+        "Environments_Https_Google" -> Environments.Https.Google
+        else -> throw IllegalArgumentException(""${'"'}No sealed enum constant ${'$'}name""${'"'})
+    }
+
+    override fun sealedObjectToEnum(obj: Environments): EnvironmentsEnum = when (obj) {
+        Environments.Http.Livefront -> EnvironmentsEnum.Environments_Http_Livefront
+        Environments.Http.Google -> EnvironmentsEnum.Environments_Http_Google
+        Environments.Https.Livefront -> EnvironmentsEnum.Environments_Https_Livefront
+        Environments.Https.Google -> EnvironmentsEnum.Environments_Https_Google
+    }
+
+    override fun enumToSealedObject(enum: EnvironmentsEnum): Environments = when (enum) {
+        EnvironmentsEnum.Environments_Http_Livefront -> Environments.Http.Livefront
+        EnvironmentsEnum.Environments_Http_Google -> Environments.Http.Google
+        EnvironmentsEnum.Environments_Https_Livefront -> Environments.Https.Livefront
+        EnvironmentsEnum.Environments_Https_Google -> Environments.Https.Google
+    }
+}
+
+/**
+ * The index of [this] in the values list.
+ */
+val Environments.ordinal: Int
+    get() = EnvironmentsSealedEnum.ordinalOf(this)
+
+/**
+ * The name of [this] for use with valueOf.
+ */
+val Environments.name: String
+    get() = EnvironmentsSealedEnum.nameOf(this)
+
+/**
+ * A list of all [Environments] objects.
+ */
+val Environments.Companion.values: List<Environments>
+    get() = EnvironmentsSealedEnum.values
+
+/**
+ * Returns an implementation of [SealedEnum] for the sealed class [Environments]
+ */
+val Environments.Companion.sealedEnum: EnvironmentsSealedEnum
+    get() = EnvironmentsSealedEnum
+
+/**
+ * Returns the [Environments] object for the given [name].
+ *
+ * If the given name doesn't correspond to any [Environments], an [IllegalArgumentException] will be
+ * thrown.
+ */
+fun Environments.Companion.valueOf(name: String): Environments =
+        EnvironmentsSealedEnum.valueOf(name)
+
+""".trimIndent()
```

Comment:
Just out of curiosity, is `trimIndent()` actually needed on any of these, since you don't have any overall indent?

Reply:
It is, if `trimIndent` is removed, these strings have an extra line at the top and bottom. It would be pretty easy to fix that, but I think it would look a bit uglier if the `package ...` line had to come directly after `"""`. Right now it's nice and clear that if you copied everything between the `"""`s that is the file that would be generated.

Reply:
Ahh that makes sense, thanks. Definitely just leave it the way it is, I just wanted to make sure it was strictly necessary 👍